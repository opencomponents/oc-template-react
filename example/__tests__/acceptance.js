jasmine.DEFAULT_TIMEOUT_INTERVAL = 100000;

const server = require("../ssr-server");
const oc = require("oc");
const packageComponent = require("oc/src/cli/domain/package-components.js");
const path = require("path");
const r = require("request-promise-native");
const JSDOM = require("jsdom").JSDOM;

const registryPort = 3000;
const registryUrl = `http://localhost:${registryPort}/`;
const serverPort = 4000;
const serverUrl = `http://localhost:${serverPort}/`;
const componentPath = path.join(__dirname, "../react-app");

let registry;
let ssrServer;

beforeAll(done => {
  registry = new oc.Registry({
    local: true,
    discovery: true,
    verbosity: 1,
    path: path.resolve("./example"),
    port: registryPort,
    baseUrl: registryUrl,
    env: { name: "local" },
    templates: [require("../../packages/oc-template-react")]
  });

  registry.start(err => {
    if (err) {
      return done(err);
    }
    ssrServer = server(serverPort, err => {
      if (err) {
        return done(err);
      }
      done();
    });
  });
});

afterAll(done => {
  ssrServer.close(() => {
    registry.close(() => {
      done();
    });
  });
});

test("Registry should correctly serve rendered and unrendered components", done => {
  const rendered = r(registryUrl + `react-app/?name=SuperMario`)
    .then(function(body) {
      expect(body).toMatchSnapshot();
    })
    .catch(err => expect(err).toBeNull());

  const unrendered = r({
    uri: registryUrl + `react-app/?name=SuperMario`,
    headers: {
      Accept: "application/vnd.oc.unrendered+json"
    }
  })
    .then(function(body) {
      expect(body).toMatchSnapshot();
    })
    .catch(err => expect(err).toBeNull());

  Promise.all([rendered, unrendered])
    .then(done)
    .catch(err => done(err));
});

test("server-side-side rendering", done => {
  JSDOM.fromURL(serverUrl + `?name=SuperMario`, {})
    .then(dom => {
      expect(dom.serialize()).toMatchSnapshot();
      done();
    })
    .catch(err => {
      expect(err).toBeNull();
    });
});
