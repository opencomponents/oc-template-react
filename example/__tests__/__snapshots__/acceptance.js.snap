// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registry should correctly serve rendered and unrendered components 1`] = `"{\\"type\\":\\"oc-component-local\\",\\"version\\":\\"1.0.0\\",\\"requestVersion\\":\\"\\",\\"name\\":\\"react-app\\",\\"renderMode\\":\\"unrendered\\",\\"href\\":\\"http://localhost:3000/react-app?name=SuperMario\\",\\"data\\":{\\"reactComponent\\":{\\"key\\":\\"4622357d373e786649bf2b755e46b8fc7d5f03b6\\",\\"src\\":\\"http://localhost:3000/react-app/1.0.0/static/react-component.js\\",\\"props\\":{\\"name\\":\\"SuperMario\\",\\"staticPath\\":\\"http://localhost:3000/react-app/1.0.0/static/\\",\\"baseUrl\\":\\"http://localhost:3000/\\"}}},\\"template\\":{\\"src\\":\\"http://localhost:3000/react-app/1.0.0/static/template.js\\",\\"type\\":\\"oc-template-react\\",\\"key\\":\\"f8718f7e66f543af3d765da26098a11bb550dac3\\"}}"`;

exports[`Registry should correctly serve rendered and unrendered components 2`] = `"{\\"type\\":\\"oc-component-local\\",\\"version\\":\\"1.0.0\\",\\"requestVersion\\":\\"\\",\\"name\\":\\"react-app\\",\\"renderMode\\":\\"rendered\\",\\"href\\":\\"http://localhost:3000/react-app?name=SuperMario\\",\\"html\\":\\"<oc-component href=\\\\\\"http://localhost:3000/react-app?name=SuperMario\\\\\\" data-hash=\\\\\\"f8718f7e66f543af3d765da26098a11bb550dac3\\\\\\" data-name=\\\\\\"react-app\\\\\\" data-rendered=\\\\\\"true\\\\\\" data-version=\\\\\\"1.0.0\\\\\\"><div id=\\\\\\"20afb707-be0d-46f2-951a-6127ceaaebe4\\\\\\"><div class=\\\\\\"oc__example-react-app-styles-css__special__3K6pmcHx\\\\\\" data-reactroot=\\\\\\"\\\\\\" data-reactid=\\\\\\"1\\\\\\" data-react-checksum=\\\\\\"706765627\\\\\\"><h1 data-reactid=\\\\\\"2\\\\\\"><!-- react-text: 3 -->Hello <!-- /react-text --><!-- react-text: 4 -->SuperMario<!-- /react-text --><!-- react-text: 5 --> <!-- /react-text --><!-- react-text: 6 -->0<!-- /react-text --></h1></div></div>\\\\n          <style>.oc__example-react-app-styles-css__special__3K6pmcHx {\\\\n  background: palevioletred;\\\\n  color: white;\\\\n}</style>\\\\n          <script>\\\\n            window.oc = window.oc || {};\\\\n            oc.cmd = oc.cmd || [];\\\\n            oc.cmd.push(function(oc){\\\\n              oc.requireSeries([{\\\\\\"global\\\\\\":\\\\\\"React\\\\\\",\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/react@15.6.2/dist/react.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"react\\\\\\"},{\\\\\\"global\\\\\\":\\\\\\"ReactDOM\\\\\\",\\\\\\"url\\\\\\":\\\\\\"https://unpkg.com/react-dom@15.6.2/dist/react-dom.min.js\\\\\\",\\\\\\"name\\\\\\":\\\\\\"react-dom\\\\\\"}], function(){\\\\n                oc.require(\\\\n                  ['oc', 'reactComponents', '4622357d373e786649bf2b755e46b8fc7d5f03b6'],\\\\n                  'http://localhost:3000/react-app/1.0.0/static/react-component.js',\\\\n                  function(ReactComponent){\\\\n                    var targetNode = document.getElementById(\\\\\\"20afb707-be0d-46f2-951a-6127ceaaebe4\\\\\\");\\\\n                    targetNode.setAttribute(\\\\\\"id\\\\\\",\\\\\\"\\\\\\");\\\\n                    ReactDOM.render(React.createElement(ReactComponent, {\\\\\\"name\\\\\\":\\\\\\"SuperMario\\\\\\",\\\\\\"staticPath\\\\\\":\\\\\\"http://localhost:3000/react-app/1.0.0/static/\\\\\\",\\\\\\"baseUrl\\\\\\":\\\\\\"http://localhost:3000/\\\\\\"}),targetNode);\\\\n                  }\\\\n                );\\\\n              });\\\\n            });\\\\n          </script>\\\\n        <script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\\\\\"react-app\\\\\\"]=\\\\\\"1.0.0\\\\\\";</script></oc-component>\\"}"`;

exports[`server-side-side rendering 1`] = `
"<!DOCTYPE html><html><head>
            <title>A page</title>
          </head>
          <body>
          <oc-component href=\\"http://localhost:3000/react-app?name=SuperMario\\" data-hash=\\"f8718f7e66f543af3d765da26098a11bb550dac3\\" data-name=\\"react-app\\" data-rendered=\\"true\\" data-version=\\"1.0.0\\"><div id=\\"20afb707-be0d-46f2-951a-6127ceaaebe4\\"><div class=\\"oc__example-react-app-styles-css__special__3K6pmcHx\\" data-reactroot=\\"\\" data-reactid=\\"1\\" data-react-checksum=\\"706765627\\"><h1 data-reactid=\\"2\\"><!-- react-text: 3 -->Hello <!-- /react-text --><!-- react-text: 4 -->SuperMario<!-- /react-text --><!-- react-text: 5 --> <!-- /react-text --><!-- react-text: 6 -->0<!-- /react-text --></h1></div></div>
          <style>.oc__example-react-app-styles-css__special__3K6pmcHx {
  background: palevioletred;
  color: white;
}</style>
          <script>
            window.oc = window.oc || {};
            oc.cmd = oc.cmd || [];
            oc.cmd.push(function(oc){
              oc.requireSeries([{\\"global\\":\\"React\\",\\"url\\":\\"https://unpkg.com/react@15.6.2/dist/react.min.js\\",\\"name\\":\\"react\\"},{\\"global\\":\\"ReactDOM\\",\\"url\\":\\"https://unpkg.com/react-dom@15.6.2/dist/react-dom.min.js\\",\\"name\\":\\"react-dom\\"}], function(){
                oc.require(
                  ['oc', 'reactComponents', '4622357d373e786649bf2b755e46b8fc7d5f03b6'],
                  'http://localhost:3000/react-app/1.0.0/static/react-component.js',
                  function(ReactComponent){
                    var targetNode = document.getElementById(\\"20afb707-be0d-46f2-951a-6127ceaaebe4\\");
                    targetNode.setAttribute(\\"id\\",\\"\\");
                    ReactDOM.render(React.createElement(ReactComponent, {\\"name\\":\\"SuperMario\\",\\"staticPath\\":\\"http://localhost:3000/react-app/1.0.0/static/\\",\\"baseUrl\\":\\"http://localhost:3000/\\"}),targetNode);
                  }
                );
              });
            });
          </script>
        <script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\"react-app\\"]=\\"1.0.0\\";</script></oc-component>
          <script>window.oc = window.oc || {};
oc.conf = oc.conf || {};
oc.conf.templates = oc.conf.templates || [];
oc.conf.templates = oc.conf.templates.concat([{\\"type\\":\\"oc-template-jade\\",\\"version\\":\\"6.0.9\\",\\"externals\\":[{\\"global\\":\\"jade\\",\\"url\\":\\"https://unpkg.com/jade-legacy@1.11.1/runtime.js\\",\\"name\\":\\"jade\\"}]},{\\"type\\":\\"oc-template-handlebars\\",\\"version\\":\\"6.0.7\\",\\"externals\\":[{\\"global\\":\\"Handlebars\\",\\"url\\":\\"https://cdnjs.cloudflare.com/ajax/libs/handlebars.js/4.0.6/handlebars.runtime.min.js\\",\\"name\\":\\"handlebars\\"}]},{\\"type\\":\\"oc-template-react\\",\\"version\\":\\"1.0.1\\",\\"externals\\":[{\\"global\\":\\"React\\",\\"url\\":\\"https://unpkg.com/react@15.6.2/dist/react.min.js\\",\\"name\\":\\"react\\"},{\\"global\\":\\"ReactDOM\\",\\"url\\":\\"https://unpkg.com/react-dom@15.6.2/dist/react-dom.min.js\\",\\"name\\":\\"react-dom\\"}]}]);</script><script src=\\"http://localhost:3000/oc-client/0.41.12/static/src/oc-client.min.js\\" type=\\"text/javascript\\"></script><script>window.oc=window.oc||{};oc.renderedComponents=oc.renderedComponents||{};oc.renderedComponents[\\"oc-client\\"]=\\"0.41.12\\";</script>
          
        
      </body></html>"
`;
